src/commands/env/var/unset.ts

```diff
@@ -52,29 +53,62 @@ export default class ConfigUnset extends Command {
       this.warn(messages.getMessage('flags.environment.deprecation'));
     }
 
-    const appName = await resolveAppNameForEnvironment(targetCompute);
-
     if (argv.length === 0) {
       throw new Errors.CLIError('you must enter a config var key (i.e. mykey)');
     }
 
-    const configPairs = argv.reduce((acc, elem) => {
-      return {
-        ...acc,
-        [elem]: null,
-      };
-    }, {});
+    try {
+      const appName = await resolveAppNameForEnvironment(targetCompute);
+
+      if (flags.json) {
+        const configPairs = argv.reduce((acc, elem) => {
+          return {
+            ...acc,
+            [elem]: null,
+          };
+        }, {});
+
+        await this.client.patch(`/apps/${appName}/config-vars`, {
+          data: configPairs,
+        });
 
-    cli.action.start(
-      `Unsetting ${Object.keys(configPairs)
-        .map((key) => herokuColor.configVar(key))
-        .join(', ')} and restarting ${herokuColor.app(targetCompute)}`
-    );
+        cli.styledJSON({
+          status: 0,
+          result: null,
+          warnings: [],
+        });
+        return;
+      } else {
+        const configPairs = argv.reduce((acc, elem) => {
+          return {
+            ...acc,
+            [elem]: null,
+          };
+        }, {});
 
-    await this.client.patch(`/apps/${appName}/config-vars`, {
-      data: configPairs,
-    });
+        cli.action.start(
+          `Unsetting ${Object.keys(configPairs)
+            .map((key) => herokuColor.configVar(key))
+            .join(', ')} and restarting ${herokuColor.app(targetCompute)}`
+        );
 
-    cli.action.stop();
+        await this.client.patch(`/apps/${appName}/config-vars`, {
+          data: configPairs,
+        });
+
+        cli.action.stop();
+      }
+    } catch (err: any) {
```

Comment:
It appears that this catch block will activate and log out JSON regardless of JSON flag being present. However, I don't even know if we want to do this - this error case should be caught by our plugin-wide error handler (which we will eventually update to include json)